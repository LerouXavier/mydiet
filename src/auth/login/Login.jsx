import React, {useState} from 'react';
import PropTypes from 'prop-types';

import {isEmpty as _isEmpty} from 'lodash';

import {Form, Message} from 'semantic-ui-react';

export const Login = () => {
    const [credentials, setCredentials] = useState({
        email: '',
        password: ''
    });

    return (
        <Form error={true}>
            <Form.Input
                className="nu-login__email"
                label="E-mail"
                onChange={(event, element) => setCredentials({email: element.value})}
                type="email"
            />
            <Form.Input
                className="nu-login__password"
                label="Password"
                onChange={(event, element) => setCredentials({password: element.value})}
                type="password"
            />
            <Form.Button
                className="nu-login__submit"
                disabled={false}
                floated="right"
                positive={true}
            >
                Login
            </Form.Button>
        </Form>
    );
}