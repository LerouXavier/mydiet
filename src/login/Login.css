#myVideo {
    position: fixed;
    right: 0;
    bottom: 0;
    min-width: 100%;
    min-height: 100%;
}

.np-login {
    height: 100vh;
}
.np-login__form {
    padding: 50px;
}
.np-login__title {
    font-family: 'Playfair Display', serif;
    font-style: italic;
    font-weight: bold;
    font-size: 54px;
    margin-bottom: 37px;
    margin-top: 23px;
}