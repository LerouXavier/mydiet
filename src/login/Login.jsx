import * as React from 'react';
import {Redirect} from 'react-router-dom';
import {isEmpty as _isEmpty} from 'lodash';

import {Form, Header, Grid, Segment} from 'semantic-ui-react';

import {authenticationService} from '../auth/AuthenticationService';
import video from './Bokeh Video Of Leaves.mp4';
import './Login.css';

export class Login extends React.Component {
    state = {
        email: '',
        password: ''
    };

    render() {
        // const {location, user} = this.props;
        // const {from} = location.state || {from: {pathname: '/'}};
        //
        // if (user !== null) {
        //     return <Redirect to={from}/>;
        // }

        return (
            <React.Fragment>
                <video autoPlay muted loop id="myVideo">
                    <source src={video} type="video/mp4"/>
                </video>

                <Grid centered={true} verticalAlign="middle" columns={4} stretched={true} className="np-login">
                    <Grid.Column>
                        <Segment as={Form} basic={true} inverted={true} padded="very" className="np-login__form np-form">
                            <Header as="h1" icon={true} textAlign="center" inverted={true}>
                                <Header.Content className="np-login__title">Nutritional Plan</Header.Content>
                            </Header>
                            <Form.Input
                                label="Wprowaź adres E-mail"
                                type="email"
                                onChange={(event, element) => this.setState({email: element.value})}
                            />
                            <Form.Input
                                label="Hasło"
                                type="password"
                                onChange={(event, element) => this.setState({password: element.value})}
                            />
                            <Form.Button
                                onClick={() => authenticationService.login(this.state)}
                                disabled={_isEmpty(this.state.email) || _isEmpty(this.state.password)}
                                positive={true}
                                floated="right"
                            >
                                Sign In
                            </Form.Button>
                        </Segment>
                    </Grid.Column>
                </Grid>
            </React.Fragment>
        );
    }
}