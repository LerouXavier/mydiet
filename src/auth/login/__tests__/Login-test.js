import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Login} from '../Login';

Enzyme.configure({adapter: new Adapter()});

describe('LoginComponent', function () {

    test('should render correctly', function () {
        const login = renderer.create(<Login/>).toJSON();
        expect(login).toMatchSnapshot();
    });

    test('should change the state after setting email', function () {
        const login = shallow(<Login/>);

        login
            .find('.nu-login__email')
            .simulate('change', {}, {value: 'example@ex.com'});

        expect(login.instance().state.email).toEqual('example@ex.com');
    });

    test('should change the state after setting password', function () {
        const login = shallow(<Login/>);

        login
            .find('.nu-login__password')
            .simulate('change', {}, {value: 'pass123'});

        expect(login.instance().state.password).toEqual('pass123');
    });

    test('should Login button be disabled while email and password are empty', function () {
        const login = shallow(<Login/>);

        expect(login.find('.nu-login__submit').prop('disabled')).toBeTruthy();
    });

    test('should enable Login button if email and password are provided', function () {
        const login = shallow(<Login/>);

        login
            .find('.nu-login__email')
            .simulate('change', {}, {value: 'example@ex.com'});

        expect(login.find('.nu-login__submit').prop('disabled')).toBeTruthy();

        login
            .find('.nu-login__password')
            .simulate('change', {}, {value: 'pass123'});


        expect(login.find('.nu-login__submit').prop('disabled')).toBeFalsy();
    });

    test('should run submitCredentials with one object attribute with credentials', function () {
        const submitCredentialsMock = jest.fn();
        const login                 = shallow(<Login submitCredentials={submitCredentialsMock}/>);

        login
            .find('.nu-login__email')
            .simulate('change', {}, {value: 'example@ex.com'});

        login
            .find('.nu-login__password')
            .simulate('change', {}, {value: 'pass123'});

        login
            .find('.nu-login__submit')
            .simulate('click');

        expect(submitCredentialsMock)
            .toBeCalledWith({email: 'example@ex.com', password: 'pass123'});
    });

    test('should display error message when credentials are incorrect', function () {
        const login = renderer.create(
            <Login
                errorMessage={{
                    header: 'Wrong credentials',
                    content: 'The credentials you supplied were not correct or did not grant access to this resource.'
                }}
            />
        ).toJSON();
        expect(login).toMatchSnapshot();
    });
});