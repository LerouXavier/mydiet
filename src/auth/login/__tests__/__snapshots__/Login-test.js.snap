// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent should display error message when credentials are incorrect 1`] = `
<form
  action={undefined}
  className="ui error form"
  onSubmit={[Function]}
>
  <div
    className="field nu-login__email"
  >
    <label
      htmlFor={undefined}
    >
      Wprowaź adres E-mail
    </label>
    <div
      className="ui input"
      content={undefined}
    >
      <input
        disabled={undefined}
        onChange={[Function]}
        required={undefined}
        tabIndex={undefined}
        type="email"
      />
    </div>
  </div>
  <div
    className="field nu-login__password"
  >
    <label
      htmlFor={undefined}
    >
      Hasło
    </label>
    <div
      className="ui input"
      content={undefined}
    >
      <input
        disabled={undefined}
        onChange={[Function]}
        required={undefined}
        tabIndex={undefined}
        type="password"
      />
    </div>
  </div>
  <div
    className="ui error message"
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Wrong credentials
      </div>
      <p>
        The credentials you supplied were not correct or did not grant access to this resource.
      </p>
    </div>
  </div>
  <div
    className="disabled field nu-login__submit"
  >
    <button
      className="ui positive disabled right floated button"
      disabled={true}
      onClick={[Function]}
      required={undefined}
      role="button"
      tabIndex={-1}
      type={undefined}
    >
      Zaloguj
    </button>
  </div>
</form>
`;

exports[`LoginComponent should render correctly 1`] = `
<form
  action={undefined}
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="field nu-login__email"
  >
    <label
      htmlFor={undefined}
    >
      Wprowaź adres E-mail
    </label>
    <div
      className="ui input"
      content={undefined}
    >
      <input
        disabled={undefined}
        onChange={[Function]}
        required={undefined}
        tabIndex={undefined}
        type="email"
      />
    </div>
  </div>
  <div
    className="field nu-login__password"
  >
    <label
      htmlFor={undefined}
    >
      Hasło
    </label>
    <div
      className="ui input"
      content={undefined}
    >
      <input
        disabled={undefined}
        onChange={[Function]}
        required={undefined}
        tabIndex={undefined}
        type="password"
      />
    </div>
  </div>
  <div
    className="disabled field nu-login__submit"
  >
    <button
      className="ui positive disabled right floated button"
      disabled={true}
      onClick={[Function]}
      required={undefined}
      role="button"
      tabIndex={-1}
      type={undefined}
    >
      Zaloguj
    </button>
  </div>
</form>
`;
